From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cogan Ng <cogan@coganng.com>
Date: Thu, 2 May 2024 20:45:47 +0800
Subject: [PATCH] Fix chat signing


diff --git a/src/main/java/net/minecraft/network/chat/SignedMessageChain.java b/src/main/java/net/minecraft/network/chat/SignedMessageChain.java
index 67eeb39aede6908d2756e49821ca350ebe916902..1a81ece67d97458a6fa845367cb2edb44eeee733 100644
--- a/src/main/java/net/minecraft/network/chat/SignedMessageChain.java
+++ b/src/main/java/net/minecraft/network/chat/SignedMessageChain.java
@@ -33,25 +33,8 @@ public class SignedMessageChain {
         SignatureValidator signatureValidator = playerPublicKey.createSignatureValidator();
         return (signature, body) -> {
             SignedMessageLink signedMessageLink = this.advanceLink();
-            if (signedMessageLink == null) {
-                throw new SignedMessageChain.DecodeException(Component.translatable("chat.disabled.chain_broken"), false); // Paper - diff on change (if disconnects, need a new kick event cause)
-            } else if (playerPublicKey.data().hasExpired()) {
-                throw new SignedMessageChain.DecodeException(Component.translatable("chat.disabled.expiredProfileKey"), false, org.bukkit.event.player.PlayerKickEvent.Cause.EXPIRED_PROFILE_PUBLIC_KEY); // Paper - kick event causes
-            } else if (!gg.pufferfish.pufferfish.PufferfishConfig.disableOutOfOrderChat && body.timeStamp().isBefore(this.lastTimeStamp)) { // Pufferfish
-                throw new SignedMessageChain.DecodeException(Component.translatable("multiplayer.disconnect.out_of_order_chat"), true, org.bukkit.event.player.PlayerKickEvent.Cause.OUT_OF_ORDER_CHAT); // Paper - kick event causes
-            } else {
-                this.lastTimeStamp = body.timeStamp();
-                PlayerChatMessage playerChatMessage = new PlayerChatMessage(signedMessageLink, signature, body, null, FilterMask.PASS_THROUGH);
-                if (!playerChatMessage.verify(signatureValidator)) {
-                    throw new SignedMessageChain.DecodeException(Component.translatable("multiplayer.disconnect.unsigned_chat"), true, org.bukkit.event.player.PlayerKickEvent.Cause.UNSIGNED_CHAT); // Paper - kick event causes
-                } else {
-                    if (playerChatMessage.hasExpiredServer(Instant.now())) {
-                        LOGGER.warn("Received expired chat: '{}'. Is the client/server system time unsynchronized?", body.content());
-                    }
-
-                    return playerChatMessage;
-                }
-            }
+            PlayerChatMessage playerChatMessage = new PlayerChatMessage(signedMessageLink, signature, body, null, FilterMask.PASS_THROUGH);
+            return playerChatMessage;
         };
     }
 
@@ -89,11 +72,7 @@ public class SignedMessageChain {
     public interface Decoder {
         static SignedMessageChain.Decoder unsigned(UUID sender, BooleanSupplier secureProfileEnforced) {
             return (signature, body) -> {
-                if (secureProfileEnforced.getAsBoolean()) {
-                    throw new SignedMessageChain.DecodeException(Component.translatable("chat.disabled.missingProfileKey"), false);
-                } else {
-                    return PlayerChatMessage.unsigned(sender, body.content());
-                }
+                return PlayerChatMessage.unsigned(sender, body.content());
             };
         }
 
