From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Cogan Ng <cogan@coganng.com>
Date: Thu, 2 May 2024 13:30:00 +0800
Subject: [PATCH] Fix game packet listener implementation


diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index de8e1656a54beff8ae37f90ae1c75dcfd6ab937c..af9ecda2134a93fe2ca1f36199138b98cf021b14 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -2280,10 +2280,6 @@ public class ServerGamePacketListenerImpl extends ServerCommonPacketListenerImpl
         synchronized (this.lastSeenMessages) {
             Optional<LastSeenMessages> optional = this.lastSeenMessages.applyUpdate(acknowledgment);
 
-            if (optional.isEmpty()) {
-                ServerGamePacketListenerImpl.LOGGER.warn("Failed to validate message acknowledgements from {}", this.player.getName().getString());
-            }
-
             return optional;
         }
     }
